### PR TITLE
Add explicit errors for failures to set_path

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously `mc-sgx-ql::PathInitializer::with_paths` returned a generic
SGX InvalidParameter error when a parameter was invalid.  Now a more
specific error is returned saying what was wrong with the parameter.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #158** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/158" target="_blank">Graphite</a> 
    * **PR #159** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/159" target="_blank">Graphite</a> 
      * **PR #163** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/163" target="_blank">Graphite</a> 
        * **PR #164** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/164" target="_blank">Graphite</a> 
          * **PR #165** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/165" target="_blank">Graphite</a> 
            * **PR #166** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/166" target="_blank">Graphite</a> 
              * **PR #171** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/171" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/171?utm_source=stack-comment).

**Review by jcape — APPROVED** [review]

**Review by awygle — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/171?utm_source=stack-comment).


Job Reference: 5plz6eqSsqN8GiebVIMp

**Pull request merged**
